src/app/components/shared/formly/image-input.component.ts

```diff
@@ -30,18 +37,55 @@ import { asFormControl } from "./helper";
         />
 
         <button
+          *ngIf="hasImage"
           type="button"
           (click)="removeImage()"
           class="btn btn-outline-danger"
-        >Remove</button>
+        >
+          Remove
+        </button>
       </div>
     </div>
   `,
 })
-export class ImageInputComponent extends FieldType {
+export class ImageInputComponent
+  extends FieldType
+  implements AfterViewInit, OnInit
+{
   @ViewChild("imageInput")
   public imageInput: ElementRef;
   public asFormControl = asFormControl;
+
+  public ngOnInit() {
+    // since the fallback image will always have the default attribute
+    // we need to check if there is more than one image with the default attribute set
+    // if the the image is the default image, update the model to use null as the image.
+    // we use an additional view model attribute to prevent sending duplicate image information in the PATCH request
+    this.model.hasDefaultImage = this.model.imageUrls.filter((image) => image.default === true).length > 1;
```

Comment:
🤔 this tricky condition makes me think we haven't set our flags right...

Reply:
also why is a component updating a model about with knowledge the model has about itself?

Reply:
okay the condition here should be: if the number of images == the number of default images, then there are only default images.

Reply:
and make this check a predicate that accepts the model to check. Don't mutate the model object